### ISSUE TITLE
feat(rules): disallow img[role] with empty alt (including role=presentation/none)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3641]
## Summary

An `<img>` element with a `role` attribute must not have an `alt` attribute whose value is the empty string.

## Spec reference

- https://w3c.github.io/html-aria/#el-img

## Impact

~1 missed error in the nu-html-checker compatibility benchmark.

**Comment by YusukeHirao** [comment]
## nu-html-checker 実出力での検証

Docker 経由で nu-html-checker を実行して確認:

### テスト HTML
```html
<img src="x.png" role="presentation" alt="">
<img src="y.png" role="img" alt="">
```

### nu-html-checker 出力
2 件の error:
> An "img" element with a "role" attribute must not have an "alt" attribute whose value is the empty string.

### markuplint 出力
1 件の error のみ:
- `role="img"` → `wai-aria-permitted-roles` で検出（img 要素に img ロール上書き不可）
- `role="presentation"` + `alt=""` のケースは **完全に漏れ**

### 判定: CONFIRMED

`role="presentation"` + `alt=""` のケースは現状どのルールでも検出されない。新しい targeted なチェックが必要。

**Comment by YusukeHirao** [comment]
## ⚠️ 過去コメントは未検証

本 Issue の 2026-04-14 以前のコメントは nu-html-checker を実行せずに書かれた推測ベースです。信頼できません。以下が Docker 経由で `ghcr.io/validator/validator:latest` を実行した実測結果です。

## 実測結果（2026-04-14）

### 全 7 ケース対比

| # | HTML | nu-html-checker | markuplint | Gap |
|---|------|----------------|-----------|-----|
| 1 | `role="presentation" alt=""` | `An "img" element with a "role" attribute must not have an "alt" attribute whose value is the empty string.` | （検出なし） | **YES** |
| 2 | `role="img" alt=""` | 同上 | 「ロール「img」を要素「img」に上書きすることはできません」(`wai-aria-permitted-roles`) | **PARTIAL**（理由違い） |
| 3 | `role="none" alt=""` | 同上 | （検出なし） | **YES** |
| 4 | `role="button" alt=""` | 同上 | 「ロール「button」...」(`wai-aria-permitted-roles`) | **PARTIAL** |
| 5 | `role="link" alt=""` | 同上 | 「ロール「link」...」(`wai-aria-permitted-roles`) | **PARTIAL** |
| 6 | `alt=""` (role なし) | 違反なし | 違反なし | NO |
| 7 | `role="presentation"` (alt なし) | `An "img" element with a "role" attribute must also have an accessible name (e.g., an "alt" attribute).` | 「ロール「presentation」...」(`wai-aria-permitted-roles`) | **PARTIAL**（意味的に衝突） |

## 真のスコープ

- **完全漏れ（新規実装必要）**: ケース 1, 3 — `role="presentation"` / `role="none"` + `alt=""` は `wai-aria-permitted-roles` が発火しないため検出されない
- **理由違い（メッセージ品質）**: ケース 2, 4, 5 — 検出はされるが「alt 空」ではなく「role 上書き不可」として報告
- **仕様解釈の衝突**: ケース 7 — nu と markuplint で判定の根拠が異なる

## 実装方針

新チェック: 条件 = `<img>` + `role` 属性あり + `alt=""` → 専用メッセージで error。

## 参考

- 検証レポート: [v5x-detailed-verification.md](https://github.com/markuplint/markuplint/issues/3684#issuecomment-4248565051)
- 再現コマンド: `docker run --rm -i ghcr.io/validator/validator:latest vnu --format json - < test.html`
